### PR TITLE
33 update use http servlet request supplier

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**
closing #33 

Review and merge #32 first.

### CODE REVIEW

**Review by nilsbehlen — COMMENTED** [review]
Does this still work with shib 4.x.x?

**Comment by lukasmatusiewicz** [comment]
yes of course

File: privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java [hunk 1]
```diff
@@ -20,88 +21,78 @@ public PrivacyIDEAAuthenticator() {}
     @Override
     protected final void doExecute(@Nonnull ProfileRequestContext profileRequestContext, @Nonnull PIContext piContext, @Nonnull PIServerConfigContext piServerConfigContext)
     {
-        HttpServletRequest request = this.getHttpServletRequest();
+        HttpServletRequest request = Objects.requireNonNull(this.getHttpServletRequestSupplier()).get();
         Map<String, String> headers = this.getHeadersToForward(request);
-        if (request == null)
+
+        piContext.setMode(request.getParameterValues("mode")[0]);
```
Comment (nilsbehlen) [inline privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java:27]: is this guaranteed to be non null?

File: privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java [hunk 2]
```diff
@@ -20,88 +21,78 @@ public PrivacyIDEAAuthenticator() {}
...
+                ActionSupport.buildEvent(profileRequestContext, "reload");
+            }
+            if (!piResponse.multichallenge.isEmpty())
+            {
+                if (debug)
                 {
-                    LOGGER.error("{} privacyIDEA server error: {}!", this.getLogPrefix(), piResponse.error.message);
-                    piContext.setFormErrorMessage(piResponse.error.message);
-                    ActionSupport.buildEvent(profileRequestContext, "reload");
+                    LOGGER.info("{} Another challenge encountered. Building form...", this.getLogPrefix());
```
Comment (nilsbehlen) [inline privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java:77]: "Challenge triggered, building form..."

File: privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java [hunk 3]
```diff
@@ -20,88 +21,78 @@ public PrivacyIDEAAuthenticator() {}
...
+            else
+            {
+                if (debug)
                 {
-                    if (debug)
-                    {
-                        LOGGER.info("{} Received a server message. Building form...", this.getLogPrefix());
-                    }
-                    ActionSupport.buildEvent(profileRequestContext, "reload");
+                    LOGGER.info("{} Received a server message. Building form...", this.getLogPrefix());
```
Comment (nilsbehlen) [inline privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java:93]: Received response from server, building form...

File: privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java [hunk 4]
```diff
@@ -20,88 +21,78 @@ public PrivacyIDEAAuthenticator() {}
     @Override
     protected final void doExecute(@Nonnull ProfileRequestContext profileRequestContext, @Nonnull PIContext piContext, @Nonnull PIServerConfigContext piServerConfigContext)
     {
-        HttpServletRequest request = this.getHttpServletRequest();
+        HttpServletRequest request = Objects.requireNonNull(this.getHttpServletRequestSupplier()).get();
```
Comment (nilsbehlen) [inline privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java:24]: I'm not really sure if an exception should be thrown if `HttpServletRequestSupplier` is null. Is is required for authentication to work, if i just want to enter my otp?

File: privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java (hunk 1 above)
Comment (lukasmatusiewicz) [reply privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java:27]: yes

File: privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java (hunk 4 above)
Comment (lukasmatusiewicz) [reply privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java:24]: All form parameters, like otp from the input field, are also transported this way.

File: privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java (hunk 3 above)
Comment (lukasmatusiewicz) [reply privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java]: fixed: [efc2703](https://github.com/privacyidea/shibboleth-plugin/pull/34/commits/efc2703e607881621baa499b2a2760c7bfebf09c)

File: privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java (hunk 2 above)
Comment (lukasmatusiewicz) [reply privacyIDEA-impl/src/main/java/org/privacyidea/action/PrivacyIDEAAuthenticator.java]: fixed: [efc2703](https://github.com/privacyidea/shibboleth-plugin/pull/34/commits/efc2703e607881621baa499b2a2760c7bfebf09c)

**Pull request merged**